[DAWN] Sorted peripherals and added Accordion

legengliu opened the pull request:
Sorted the list of peripherals alphabetically, first by type, then by name; added Accordion to the list of peripherals so now they can be collapsed and expanded.

Commit:
Sorted peripherals and added Accordion

Commit:
Sorted peripherals and added Accordion

kevinmasd committed:
[Dawn] Development also needs to pull in protos (#134)

onibrow committed:
Add instructions to update trusty crouton build (#122)

bodiddles committed:
Runtime/fix udp tests (#128)

* [RUNTIME/DAWN] Adds E-stop to Ansible proto

Added E-stop enum/command to the udp proto dawn sends to runtime

* [RUNTIME] Fixes Runtime Tests to not crash

Only terminates UDP Send process if it has spawned in the first place.
Reflects new behavior of Ansible Communication

* [RUNTIME] Fixes print statements to match expected output

* [RUNTIME] minor text fixes

Removes spaces after certain lines for style.

nikitakit committed:
[DevOps] Set up Travis to build dawn and runtime (#131)

* [DevOps] Update create_update.sh

Build the protobuf python files, and don't automatically pull from
GitHub (that prevents using this script inside travis)

* [RUNTIME] Return nonzero exit code on test failure

* [Dawn] Create build directory if it doesn't exist

* [DevOps] Set up Travis to build dawn and runtime (#131)

By default, travis will run dawn tests, runtime tests, and package
a runtime update tarball (but not do anything with it).

If a commit is tagged starting with dawn/ or test/dawn/, it will
additionally build Dawn binaries and upload them to GitHub releases.

If a commit is tagged starting with runtime/ or test/runtime, it
will additionally upload the runtime update tarball to GitHub
releases.

yizow reviewed (approved):
LGTM

Few points:
1) In future, make sure to mark someone as reviewer so they get a notification about your pull request.
2) Make sure to rebase before pushing to minimize merge conflicts (instructions in main readme of PieCentral repo)

kevinji committed:
[RUNTIME] studentAPI: Refactor/simplify code (#135)

kevinji committed:
[RUNTIME] Use requirements.txt for pip dependencies (#136)

* [RUNTIME] Use requirements.txt for pip dependencies

* [RUNTIME] requirements.txt: Add pyserial

karthik-shanmugam committed:
[HIBIKE] document usage of virtual devices (#125)

- add a markdown file documenting the usage of virtual devices
- modify `spawn_virtual_devices.py`, moving the user editable section to the top.
- change the example list of virtual devices in `spawn_virtual_devices.py`
  - demonstrate all 3 supported device types
  - demonstrate support for spawning multiple devices of the same type
- minor bugfix: invoke `python3` instead of `python3.5` , since virtual devices don't require python3.5

karthik-shanmugam committed:
[HIBIKE] runtime integration fixes (#142)

- update `hibike_process.py` to send uid's along with device values, as was written in the integration spec (fix implementation to match documentation)
- update and rename integration spec
  - add `[]`'s around arguments to accurately reflect how contents of messages are nested (fix documentation to match implementation)
  - rename ATALANTA => RUNTIME

yizow committed:
[RUNTIME] runtime.py: studentCode timing bugfix (#144)

Calculates the time for execution of the next studentCode main loop
*before* we execute the main loop.

karthik-shanmugam committed:
[HIBIKE] always store virtual device config file in hibike folder (#145)

Minor change so running this program outside PiECentral/hibike doesn't break it by writing "virtual_devices.txt" to the wrong folder.

kevinmasd committed:
[Dawn] Accordion Applies to All of Same Type

kevinmasd committed:
[Dawn] Merge Conflicts

kevinmasd commented:
Accordions were placed on individual peripherals rather than by types. Fixed with current changes.

kevinmasd committed:
[Dawn] Lint Failings

kevinmasd reviewed (approved):
Post-Rebased
